lib/internal/child_process.js

```diff
@@ -78,21 +78,26 @@ const handleConversion = {
         if (firstTime) socket.server._setupSlave(socketList);
 
         // Act like socket is detached
-        socket.server._connections--;
+        if (!options.keepOpen)
+          socket.server._connections--;
+
       }
 
+      var handle = socket._handle;
+
       // remove handle from socket object, it will be closed when the socket
       // will be sent
-      var handle = socket._handle;
-      handle.onread = function() {};
-      socket._handle = null;
+      if (!options.keepOpen) {
+        handle.onread = function() {};
+        socket._handle = null;
+      }
 
       return handle;
     },
 
-    postSend: function(handle) {
+    postSend: function(handle, options) {
       // Close the Socket handle after sending it
```

Comment:
very minor nit: wouldn't it be *slightly* more efficient to: 
```js
if (options.keepOpen) return;
if (handle) handle.close();
```
/cc @bnoordhuis